src/convertor/OndcItemMapper.spec.js

```diff
@@ -0,0 +1,174 @@
+import {
+  it, expect, describe,
+} from 'vitest';
+import platformMapperConfig from '../resource/test/expectedMapperOutput.json';
+import OndcItemMapper from './OndcItemMapper';
+
+describe('OndcItemMapper', () => {
+  it('should match ondc string to platform string ', async () => {
+    const responseWithName = {
+      name: 'XYZ',
+    };
+    const config = [{
+      ondc: 'descriptor.name',
+      ondcDataType: 'string',
+      platform: '.name',
+      platformDataType: 'string',
+    }];
+    const ondcItemMapper = new OndcItemMapper(config);
+    const mappedNameValue = await ondcItemMapper.map(responseWithName);
+    expect(mappedNameValue).toStrictEqual(
+      [
+        {
+          ondc: 'descriptor.name',
+          ondcDataType: 'string',
+          platform: '.name',
+          platformDataType: 'string',
+          platformValue: 'XYZ',
+        },
+      ],
+    );
+  });
+  it('should match ondc string to platform number ', async () => {
+    const responseWithId = {
+      id: 52,
+    };
+    const config = [
+      {
+        ondc: 'id',
+        ondcDataType: 'string',
+        platform: '.id',
+        platformDataType: 'number',
+      }];
+    const ondcItemMapper = new OndcItemMapper(config);
+    const mappedIdValue = await ondcItemMapper.map(responseWithId);
+    expect(mappedIdValue).toStrictEqual(
+      [
+        {
+          ondc: 'id',
+          ondcDataType: 'string',
+          platform: '.id',
+          platformDataType: 'number',
+          platformValue: '52',
+        },
+      ],
+    );
+  });
+  it('should match ondc nested object to platform nested object ', async () => {
```

Comment:
This testcase should have been map array to array

Reply:
updated